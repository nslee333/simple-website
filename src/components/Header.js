import Image from "next/image";
import bendjs_logo from "../../public/images/bendjs_logo_square.png";
import Head from "next/head";

export default function Header() {
  return (
    <div className="flex justify-center">
      <Head>
        <title>BendJS Website</title>
      </Head>
      <div className="mx-5 my-5 w-[50rem]">
        <a href="/">
          <Image
            className="w-[8rem] rounded-lg"
            src={bendjs_logo}
            alt="bendjs logo"
          />
        </a>
      </div>
    </div>
  );
}
