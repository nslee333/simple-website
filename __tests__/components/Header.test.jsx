import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Header from "../../src/components/Header";

test("Header renders without crashing", () => {
  render(<Header />);
});

it("Header renders correctly according to snapshot", () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
