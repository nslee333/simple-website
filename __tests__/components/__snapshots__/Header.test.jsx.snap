// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly according to snapshot 1`] = `
<div
  className="flex justify-center"
>
  <div
    className="mx-5 my-5 w-[50rem]"
  >
    <a
      href="/"
    >
      <img
        alt="bendjs logo"
        className="w-[8rem] rounded-lg"
        data-nimg="1"
        decoding="async"
        height={40}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={40}
      />
    </a>
  </div>
</div>
`;
